nrf_wifi/fw_if/umac_if/src/event.c

```diff
@@ -927,62 +927,93 @@ static enum nrf_wifi_status umac_process_sys_events(struct nrf_wifi_fmac_dev_ctx
-					def_dev_ctx->tx_config.peers[MAX_PEERS].peer_id =
-										MAX_PEERS;
-					def_dev_ctx->tx_config.peers[MAX_PEERS].if_idx =
-									mode_event->if_index;
-					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
-									NRF_WIFI_STA_TX_INJECTOR;
-					def_dev_ctx->vif_ctx[mode_event->if_index]->txinjection_mode
-									= true;
-				} else
-#endif /* CONFIG_NRF700X_RAW_DATA_TX */
-				if (mode_event->op_mode == NRF_WIFI_STA_MODE) {
-					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
-									NRF_WIFI_IFTYPE_STATION;
+				vif->txinjection_mode = false;
+#endif
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+				vif->promisc_mode = false;
+#endif
+				if ((mode_event->op_mode & NRF_WIFI_STA_MODE)
+								== NRF_WIFI_STA_MODE) {
+					mode_event->op_mode =
+						(mode_event->op_mode ^ NRF_WIFI_STA_MODE);
+					vif->if_type = NRF_WIFI_STA_MODE;
+					config->peers[MAX_PEERS].peer_id = -1;
+					config->peers[MAX_PEERS].if_idx = -1;
+
+#if defined(CONFIG_NRF700X_RAW_DATA_TX) && defined (CONFIG_NRF700X_PROMISC_DATA_RX)
+					if (mode_event->op_mode) {
+						if ((mode_event->op_mode ^
+						    (NRF_WIFI_PROMISCUOUS_MODE |
+						     NRF_WIFI_TX_INJECTION_MODE)) == 0) {
+							vif->if_type =
+								NRF_WIFI_STA_PROMISC_TX_INJECTOR;
+							config->peers[MAX_PEERS].peer_id
+									= MAX_PEERS;
+							config->peers[MAX_PEERS].if_idx
+									= if_idx;
+							vif->txinjection_mode = true;
+							vif->promisc_mode = true;
+							break;
+						}
+#endif /* CONFIG_NRF700X_RAW_DATA_TX && defined CONFIG_NRF700X_PROMISC_DATA_RX */
 #ifdef CONFIG_NRF700X_RAW_DATA_TX
-					def_dev_ctx->vif_ctx[mode_event->if_index]->txinjection_mode
-									= false;
+						if ((mode_event->op_mode ^
+						     NRF_WIFI_TX_INJECTION_MODE) == 0) {
+							config->peers[MAX_PEERS].peer_id
+									= MAX_PEERS;
+							config->peers[MAX_PEERS].if_idx
+									= if_idx;
+							vif->if_type = NRF_WIFI_STA_TX_INJECTOR;
+							vif->txinjection_mode = true;
+							break;
+						}
 #endif /* CONFIG_NRF700X_RAW_DATA_TX */
-					def_dev_ctx->tx_config.peers[MAX_PEERS].peer_id = -1;
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+						if ((mode_event->op_mode ^
+						     NRF_WIFI_PROMISCUOUS_MODE) == 0) {
+							vif->if_type = NRF_WIFI_STA_PROMISC;
+							vif->promisc_mode = true;
+							break;
+						}
+					}
+#endif /* CONFIG_NRF700X_PROMISC_DATA_RX */
 				}
 #ifdef CONFIG_NRF700X_RAW_DATA_RX
-				else if (mode_event->op_mode == NRF_WIFI_MONITOR_MODE) {
-					def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
-									NRF_WIFI_IFTYPE_MONITOR;
-					def_dev_ctx->tx_config.peers[MAX_PEERS].peer_id = -1;
+				if ((mode_event->op_mode & NRF_WIFI_MONITOR_MODE)
+							== NRF_WIFI_MONITOR_MODE) {
+					mode_event->op_mode =
+						(mode_event->op_mode ^ NRF_WIFI_MONITOR_MODE);
+					vif->if_type = NRF_WIFI_IFTYPE_MONITOR;
+					config->peers[MAX_PEERS].peer_id = -1;
+					config->peers[MAX_PEERS].if_idx = -1;
+
+					if (mode_event->op_mode) {
```

Comment:
Why is this check necessary (when we have the inner checks in place) ?

Reply:
Kindly refer previous comment and note for why this check is placed